### ISSUE TITLE
Add learning rate schedulers

### ISSUE THREAD

**Issue by jessegrabowski** [pymc-devs/pytensor_ml#6]
The learning rate should be controllable via a scheduler. Here are torch examples:

- [ ] [Reduce on Plateau](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.ReduceLROnPlateau.html#torch.optim.lr_scheduler.ReduceLROnPlateau)
- [ ] [Linear Decay](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.LinearLR.html#torch.optim.lr_scheduler.LinearLR)
- [ ] [Exponential Decay](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.ExponentialLR.html#torch.optim.lr_scheduler.ExponentialLR)
- [ ] [Cosine Annealing](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.CosineAnnealingLR.html#torch.optim.lr_scheduler.CosineAnnealingLR)

There are others, see [here ](https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate)for the full list.
